backend/app/api/auterity_expansion_fixed.py

```
@@ -0,0 +1,508 @@
+        logger.error(f"Failed to deploy agent: {str(e)}")
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"Failed to deploy agent: {str(e)}",
+        )
+
+
+@router.post("/agents/{agent_id}/tasks")
+async def assign_task_to_agent(
+    agent_id: UUID,
+    task_data: Dict[str, Any],
+    db: Session = Depends(get_db),
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)